app/code/Magento/Newsletter/Controller/Subscriber/NewAction.php

```diff
@@ -125,27 +131,26 @@ public function execute()
                 $this->validateEmailAvailable($email);
 
                 $subscriber = $this->_subscriberFactory->create()->loadByEmail($email);
-                if ($subscriber->getId()
-                    && $subscriber->getSubscriberStatus() == \Magento\Newsletter\Model\Subscriber::STATUS_SUBSCRIBED
-                ) {
-                    throw new \Magento\Framework\Exception\LocalizedException(
+                if ($subscriber->getId() && $subscriber->getSubscriberStatus() == Subscriber::STATUS_SUBSCRIBED) {
+                    throw new LocalizedException(
                         __('This email address is already subscribed.')
                     );
                 }
 
                 $status = $this->_subscriberFactory->create()->subscribe($email);
-                if ($status == \Magento\Newsletter\Model\Subscriber::STATUS_NOT_ACTIVE) {
-                    $this->messageManager->addSuccess(__('The confirmation request has been sent.'));
-                } else {
-                    $this->messageManager->addSuccess(__('Thank you for your subscription.'));
+                if ($status == Subscriber::STATUS_NOT_ACTIVE) {
+                    $this->messageManager->addSuccessMessage(__('The confirmation request has been sent.'));
```

Comment:
In this case redirect will not be executed. Currently this change adds regression. It's better to revert it or suggest another solution